### PR TITLE
Correctly dump the data of babelfish_extended_properties

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
Previously, there were two issues with the dump of `babelfish_extended_properties`
catalog table:  
1. Current restore logic requires that `babelfish_sysdatabases` table's data gets
restored first before any other table containing `dbid` column so that we generate
the `dbid` once in the beginning and use it for all the remaining tables. But it was
not the case yet.
2. `sql_variant` column handling was not there for catalog table's data but 
`babelfish_extended_properties` contains it.

This commit fixes both the above two issues so that we correctly dump the data
of `babelfish_extended_properties` catalog.

Task: BABEL-4720 
Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>

### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/bin/pg_dump/dump_babel_utils.c [hunk 1]
```diff
@@ -1237,7 +1248,7 @@ fixCopyCommand(Archive *fout, PQExpBuffer copyBuf, TableInfo *tbinfo, bool isFro
 	 * Return if not a Babelfish database, or if the table is not a Babelfish
 	 * configuration table.
 	 */
-	if (!isBabelfishDatabase(fout) || !isBabelfishConfigTable(tbinfo))
+	if (!isBabelfishDatabase(fout) || !isBabelfishConfigTable(fout, tbinfo))
```
Comment (surendravishnoi) [inline src/bin/pg_dump/dump_babel_utils.c]: do we need isBabelfishDatabase check?

File: src/bin/pg_dump/dump_babel_utils.c [hunk 2]
```diff
@@ -1380,8 +1391,9 @@ fixCursorForBbfSqlvariantTableData( Archive *fout,
 {
 	int orig_nfields = 0;
 	PQExpBuffer buf = createPQExpBuffer();
+	bool is_catalog_table = isBabelfishConfigTable(fout, tbinfo);
 
-	if (!isBabelfishDatabase(fout) || !hasSqlvariantColumn(tbinfo))
```
Comment (surendravishnoi) [inline src/bin/pg_dump/dump_babel_utils.c:1384]: why we removed isBabelfishDatabase check?

File: src/bin/pg_dump/dump_babel_utils.c [hunk 3]
```diff
@@ -1409,15 +1421,34 @@ fixCursorForBbfSqlvariantTableData( Archive *fout,
 		if (pg_strcasecmp(tbinfo->atttypnames[i],
 			quote_all_identifiers ? "\"sys\".\"sql_variant\"" : "sys.sql_variant") == 0)
 		{
-			appendPQExpBuffer(buf, ", sys.SQL_VARIANT_PROPERTY(%s, 'BaseType')", fmtId(tbinfo->attnames[i]));
-			appendPQExpBuffer(buf, ", sys.datalength(%s)", fmtId(tbinfo->attnames[i]));
+			appendPQExpBufferStr(buf, ", sys.SQL_VARIANT_PROPERTY(");
+			if (is_catalog_table)
```
Comment (surendravishnoi) [inline src/bin/pg_dump/dump_babel_utils.c:1428]: Please add a comment explain why we need to change query for catalog tables

File: src/bin/pg_dump/dump_babel_utils.c [hunk 4]
```diff
@@ -1409,15 +1421,34 @@ fixCursorForBbfSqlvariantTableData( Archive *fout,
...
 		orig_nfields++;
 	}
 	appendPQExpBufferStr(query, buf->data);
 	destroyPQExpBuffer(buf);
+
+	/*
+	 * Add the FROM clause which we delayed in fixCursorForBbfCatalogTableData function
+	 * for catalog tables.
+	 */
+	if (is_catalog_table) {
```
Comment (surendravishnoi) [inline src/bin/pg_dump/dump_babel_utils.c]: why we have different conditions to skip (sql variant) and add (catalog table) FROM clause?

File: src/bin/pg_dump/dump_babel_utils.c [hunk 5]
```diff
@@ -1171,7 +1175,7 @@ fixCursorForBbfCatalogTableData(Archive *fout, TableInfo *tbinfo, PQExpBuffer bu
 	{
 		if (tbinfo->attisdropped[i])
 			continue;
-		if (tbinfo->attgenerated[i] && fout->dopt->column_inserts)
+		if (tbinfo->attgenerated[i] && (fout->dopt->column_inserts || has_sqlv_col))
```
Comment (surendravishnoi) [inline src/bin/pg_dump/dump_babel_utils.c:1192]: why this change? how sql variant relates to generated columns

File: src/bin/pg_dump/dump_babel_utils.c (hunk 2 above)
Comment (rishabhtanwar29) [reply src/bin/pg_dump/dump_babel_utils.c:1384]: Actually it is already being checked in `fixCursorForBbfTableData` function so it is a redundant check here.

File: src/bin/pg_dump/dump_babel_utils.c (hunk 4 above)
Comment (rishabhtanwar29) [reply src/bin/pg_dump/dump_babel_utils.c]: I have moved the FROM clause logic to a helper function `addFromClauseForBabelfishCatalogTable`.

File: src/bin/pg_dump/dump_babel_utils.c (hunk 5 above)
Comment (rishabhtanwar29) [reply src/bin/pg_dump/dump_babel_utils.c:1192]: This change is taken from the existing logic in `dumpTableData_insert`.

File: src/bin/pg_dump/dump_babel_utils.c (hunk 1 above)
Comment (rishabhtanwar29) [reply src/bin/pg_dump/dump_babel_utils.c]: Yeah we don't need it now, removed it.

**Review by surendravishnoi — APPROVED** [review]
